apis/cloudbuild/v1beta1/workerpool_types.go

```diff
@@ -0,0 +1,125 @@
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/clients/generated/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudBuildWorkerPoolSpec defines the desired state of Instance
+type CloudBuildWorkerPoolSpec struct {
+	Name        string  `json:"name,omitempty"`
+	DisplayName string  `json:"displayName,omitempty"`
+	ResourceID  *string `json:"resourceID,omitempty"`
+	// +required
+	ProjectRef *refv1beta1.ProjectRef `json:"projectRef"`
+	// +required
+	Location string `json:"location"`
+	// +required
+	PrivatePoolConfig *PrivatePoolV1Config `json:"privatePoolV1Config,omitempty"`
+}
+
+type PrivatePoolV1Config struct {
+	// +required
+	WorkerConfig *WorkerConfig `json:"workerConfig,omitempty"`
+	// +optional
+	NetworkConfig *NetworkConfig `json:"networkConfig,omitempty"`
+}
+
+type WorkerConfig struct {
+	// +optional
+	MachineType string `json:"machineType,omitempty"`
+	// +optional
+	DiskSizeGb int64 `json:"diskSizeGb,omitempty"`
+}
+
+type NetworkConfig struct {
+	// +required
+	PeeredNetworkRef refv1beta1.ComputeNetworkRef `json:"peeredNetworkRef,omitempty"`
+	// +optional
+	EgressOption string `json:"egressOption,omitempty"`
+	// +optional
+	PeeredNetworkIpRange string `json:"peeredNetworkIpRange,omitempty"`
+}
+
+// CloudBuildWorkerPoolStatus defines the observed state of Instance
+type CloudBuildWorkerPoolStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	/* ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource. */
+	// +optional
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	/* The creation timestamp of the workerpool.*/
+	// +optional
+	// +kubebuilder:validation:Format=date-time
+	CreateTime *string `json:"createTime,omitempty"`
+
+	/* The last update timestamp of the workerpool.*/
+	// +optional
+	// +kubebuilder:validation:Format=date-time
+	UpdateTime *string `json:"updateTime,omitempty"`
+
+	ResourceID *string `json:"resourceID,omitempty"`
+
+	/* ObservedState is the state of the resource as most recently observed in GCP. */
+	// +optional
+	ObservedState *CloudBuildWorkerPoolObservedState `json:"observedState,omitempty"`
+}
+
+type CloudBuildWorkerPoolObservedState struct {
+	// +required
+	WorkerConfig  *WorkerConfig       `json:"workerConfig,omitempty"`
+	NetworkConfig *NetworkConfigState `json:"networkConfig,omitempty"`
+}
+
+type NetworkConfigState struct {
+	// +optional
+	PeeredNetwork string `json:"peeredNetwork,omitempty"`
+	// +optional
+	EgressOption string `json:"egressOption,omitempty"`
+	// +optional
+	PeeredNetworkIpRange string `json:"peeredNetworkIpRange,omitempty"`
```

Comment:
We should decide whether it is `peerNetworkIPRange` or `peeredNetworkIPRange`

I sent https://github.com/GoogleCloudPlatform/k8s-config-connector/pull/1953 which links to the Kube conventions guide and tries to enforce it, but the list of exceptions is so large that we might decide that we're not following it...